osf/external/gravy_valet/translations.py

```diff
@@ -284,3 +282,29 @@ def after_fork(self, node: 'Node', fork, user: 'OSFUser', save=True):
             relationships=json_data['relationships'],
             addon_type=self.gv_data.resource_type
         )
+
+def get_settings_class(addon_type):
+    if addon_type == AddonType.STORAGE:
+        return _get_storage_settings_class()
+
+    return EphemeralNodeSettings
+
+
+def _get_storage_settings_class():
+    if _StorageEphemeralNodeSettings is None:
+        _initialize_ephemeral_storage_node_settings()
+    return _StorageEphemeralNodeSettings
+
+
+def _initialize_ephemeral_storage_node_settings():
+    from addons.base.models import BaseStorageAddon
+
+    global _StorageEphemeralNodeSettings
+
+    class StorageEphemeralNodeSettings(EphemeralNodeSettings, BaseStorageAddon):
+        pass
+
+    _StorageEphemeralNodeSettings = StorageEphemeralNodeSettings
```

Comment:
I'm not happy with this approach either, so I am more than welcome to suggestions

Reply:
I'm afraid I don't have any other suggestions.

Reply:
Might avoid using a global with this:
```
def _initialize_ephemeral_storage_node_settings():
    if not hasattr(_initialize_ephemeral_storage_node_settings, "_storage_node_settings"):
        from addons.base.models import BaseStorageAddon

        class StorageEphemeralNodeSettings(EphemeralNodeSettings, BaseStorageAddon):
            pass

        _initialize_ephemeral_storage_node_settings._storage_node_settings = StorageEphemeralNodeSettings

    return _initialize_ephemeral_storage_node_settings._storage_node_settings

```
I think that cover any import problems with a global.